src/data/java_profile.rs

```diff
@@ -0,0 +1,340 @@
+        }
+
+        self.process_map.iter().find_map(|(jid, name)| {
+            if key == name[0] {
+                Some(jid.to_string())
+            } else {
+                None
+            }
+        })
+    }
+
+    fn update_process_map(&mut self) -> String {
+        debug!("Running jps (may incur utilization spike)...");
+        let jps_cmd = Command::new("jps").output();
+        let mut jps_str = String::new();
+        match jps_cmd {
+            Ok(jps_out) => {
+                jps_str = String::from_utf8(jps_out.stdout).unwrap_or_default();
+                let jps: Vec<&str> = jps_str.split_whitespace().collect();
+                for i in (0..jps.len()).step_by(2) {
+                    if jps[i + 1] != "Jps" {
+                        self.process_map
+                            .insert(String::from(jps[i]), vec![String::from(jps[i + 1])]);
+                    }
+                }
+            }
+            Err(e) => {
+                error!("Jps command failed: {}", e);
+            }
+        }
+        jps_str
+    }
+
+    fn launch_pgrep(&mut self) -> Vec<String> {
+        let pgrep_cmd = Command::new("pgrep").arg("java").output();
+        let mut pgrep: Vec<String> = Vec::new();
+        match pgrep_cmd {
+            Ok(pgrep_out) => {
+                let pgrep_str = String::from_utf8(pgrep_out.stdout).unwrap();
+                pgrep = pgrep_str
+                    .split_whitespace()
+                    .map(|s| s.to_string())
+                    .collect();
+            }
+            Err(e) => {
+                error!("Pgrep command failed: {}", e);
+            }
+        }
+        pgrep
+    }
+}
+
+impl CollectData for JavaProfileRaw {
+    fn prepare_data_collector(&mut self, params: CollectorParams) -> Result<()> {
+        let mut jids: Vec<String> = Vec::new();
+        let pgrep: Vec<String> = self.launch_pgrep();
+        for pid in pgrep {
+            if self.process_map.contains_key(&pid) {
+                continue;
+            }
+            self.process_map
+                .insert(pid.clone(), vec![String::from("Could not resolve name!")]);
+        }
+
+        let jps_str = self.update_process_map();
+        let jps: Vec<&str> = jps_str.split_whitespace().collect();
+
+        let default = String::from("jps");
+        let jprofile = params
+            .profile
+            .get(JAVA_PROFILE_FILE_NAME)
+            .unwrap_or(&default)
+            .as_str();
+        match jprofile {
+            "jps" => {
+                jids = self.process_map.clone().into_keys().collect();
+                debug!("Jps will be run if new JVM is started during aperf record to resolve process names.",);
+            }
+            _ => {
+                let args: Vec<&str> = jprofile.split(',').collect();
+                for arg in args {
+                    if !jps.contains(&arg) {
+                        error!("No JVM with name/PID '{}'.", arg);
+                        continue;
```

Comment:
We have to figure out what the behavior is here. I would say specifying a pid that doesn't exist should be fatal, but names are not as clear-cut. We may want to keep looking for only that name to appear during recording.

Reply:
What are the names, anyway? Can you provide examples?

Reply:
Names are either the classname or JAR file name that the JVM was launched on. For example, if I compile and run program.java, I can specify it with the java profile flag: `./aperf record --profile-java program`.

Reply:
When providing multiple PIDs, what if some are valid but some aren't?